### PR TITLE
provision shard generation

### PR DESCRIPTION
**Pull request by geoberle**
### What this PR does

* creation of local and in-cluster provisioning shard
* preparation for KV config addition
* new make target `local-deploy-provision-shard` to generate a provisioning shard for local development usage

Jira: 
Link to demo recording:

### CODE REVIEW

File: cluster-service/Makefile [hunk 1]
```diff
@@ -41,29 +41,33 @@ deploy-local-db-secret: configure-tmp-provision-shard
...
 	oc process --local -f deploy/integration/cluster-service-namespace.yaml \
 		-p CLIENT_ID=$${AZURE_CS_MI_CLIENT_ID} | oc apply -f -
 
-# for local development
 provision-shard:
-	ZONE_RESOURCE_ID=$(shell az network dns zone show -n ${ZONE_NAME} -g ${REGIONAL_RESOURCEGROUP} --query id -o tsv) && \
-	../templatize.sh $(DEPLOY_ENV) deploy/provisioning-shards.tmpl.yml deploy/provisioning-shards.yml -e zoneResourceId=$${ZONE_RESOURCE_ID}
+	@ZONE_RESOURCE_ID=$(shell az network dns zone show -n ${ZONE_NAME} -g ${REGIONAL_RESOURCEGROUP} --query id -o tsv) && \
+	CX_SECRETS_KV_URL=$(shell az keyvault show -n ${CX_SECRETS_KV_NAME} -g ${MGMT_RESOURCEGROUP} --query properties.vaultUri -o tsv) && \
+	CX_MI_KV_URL=$(shell az keyvault show -n ${CX_MI_KV_NAME} -g ${MGMT_RESOURCEGROUP} --query properties.vaultUri -o tsv) && \
```
Comment (JameelB) [inline cluster-service/Makefile:60]: Are these ok to be here even when its not used, its just going to be ignored?

File: cluster-service/Makefile [hunk 2]
```diff
@@ -41,29 +41,33 @@ deploy-local-db-secret: configure-tmp-provision-shard
...
+	../templatize.sh $(DEPLOY_ENV) deploy/provisioning-shards.tmpl.yml deploy/provisioning-shards.yml -e zoneResourceId=$${ZONE_RESOURCE_ID},cxSecretsKeyVaultUrl=$${CX_SECRETS_KV_URL},cxMiKeyVaultUrl=$${CX_MI_KV_URL},maestroRestUrl=http://maestro.maestro.svc.cluster.local:8000,maestroGrpUrl=maestro-grpc.maestro.svc.cluster.local:8090
 	@cat deploy/provisioning-shards.yml
 
+local-deploy-provision-shard:
+	@ZONE_RESOURCE_ID=$(shell az network dns zone show -n ${ZONE_NAME} -g ${REGIONAL_RESOURCEGROUP} --query id -o tsv) && \
+	CX_SECRETS_KV_URL=$(shell az keyvault show -n ${CX_SECRETS_KV_NAME} -g ${MGMT_RESOURCEGROUP} --query properties.vaultUri -o tsv) && \
+	CX_MI_KV_URL=$(shell az keyvault show -n ${CX_MI_KV_NAME} -g ${MGMT_RESOURCEGROUP} --query properties.vaultUri -o tsv) && \
+	../templatize.sh $(DEPLOY_ENV) deploy/provisioning-shards.tmpl.yml deploy/local-provisioning-shards.yml -e zoneResourceId=$${ZONE_RESOURCE_ID},cxSecretsKeyVaultUrl=$${CX_SECRETS_KV_URL},cxMiKeyVaultUrl=$${CX_MI_KV_URL},maestroRestUrl=http://localhost:8080,maestroGrpUrl=localhost:8090
```
Comment (JameelB) [inline cluster-service/Makefile:68]: Should be `http://localhost:8001`, which is how we configured it in the port forward cmd in the dev setup guide? 
```
kubectl port-forward svc/maestro 8001:8000 -n maestro
```

Comment (machi1990) [reply cluster-service/Makefile:68]: good catch!

File: cluster-service/Makefile (hunk 1 above)
Comment (geoberle) [reply cluster-service/Makefile:60]: yes, we can keep them here for the upcoming change @machi1990 is working on

**Pull request merged**